include/openssl/service_indicator.h

```diff
@@ -0,0 +1,33 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef AWSLC_HEADER_SERVICE_INDICATOR_H
+#define AWSLC_HEADER_SERVICE_INDICATOR_H
+
+#if defined(__cplusplus)
+extern "C" {
+#endif
+
+OPENSSL_EXPORT int awslc_fips_service_indicator_get_counter(void);
+OPENSSL_EXPORT int awslc_fips_service_indicator_get_serviceID(void);
+OPENSSL_EXPORT int awslc_fips_service_indicator_reset_state(void);
+
+#if defined(AWSLC_FIPS)
+#define IS_FIPS_APPROVED_CALL_SERVICE(approved, func, ...)                  \
```

Comment:
There should also be a macro for non-fips, makes it much easier for apps to have a uniform codebase.